packages/actor-rdf-resolve-quad-pattern-federated/lib/FederatedQuadSource.ts

```diff
@@ -149,25 +219,28 @@ export class FederatedQuadSource implements ILazyQuadSource {
     //       See discussion at https://github.com/comunica/comunica/issues/553
     const sourcesIt = this.sources.iterator();
     const proxyIt: AsyncIterator<PromiseProxyIterator<RDF.Quad>> = sourcesIt.map((source) => {
+      const sourceId = this.getSourceId(source);
       remainingSources++;
       sourcesCount++;
 
-      // If we can predict that the given source will have no bindings for the given pattern,
-      // return an empty iterator.
-      const pattern: Algebra.Pattern = this.algebraFactory.createPattern(
-        FederatedQuadSource.nullToVariable(subject, 's'),
-        FederatedQuadSource.nullToVariable(predicate, 'p'),
-        FederatedQuadSource.nullToVariable(object, 'o'),
-        FederatedQuadSource.nullToVariable(graph, 'g'),
-      );
+      return new PromiseProxyIterator(async () => {
+        // Deskolemize terms, so we send the original blank nodes to each source.
```

Comment:
Hmm, but will that work for all sources? Clearly not for SPARQL endpoints.

Reply:
Good point...

But I would keep this for another issue, since this also applied before this PR, and I don't think it's a trivial thing to fix.

Reply:
It's unfixable in general; perhaps just good to add a note.

Reply:
It's unfixable in general; perhaps just good to add a note.